src/Sylius/Bundle/CoreBundle/Tests/Controller/ShippingCategoryControllerTest.php

```diff
@@ -0,0 +1,207 @@
+        ]);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'error/not_found', 404);
+    }
+
+    public function testFullUpdateShippingCategoryValidationFailResponse()
+    {
+        $this->resetDataBase();
+        $id = $this->createShippingCategory();
+
+        $this->client->request('PUT', '/api/shipping-categories/'.$id, [], [], [
+            'HTTP_Authorization' => 'Bearer access_token',
+            'CONTENT_TYPE' => 'application/json',
+        ]);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'shipping_category/validation_fail', 400);
+    }
+
+    public function testFullUpdateShippingCategoryResponse()
+    {
+        $this->resetDataBase();
+        $id = $this->createShippingCategory();
+
+        $this->client->request('PUT', '/api/shipping-categories/'.$id, [], [], [
+            'HTTP_Authorization' => 'Bearer access_token',
+            'CONTENT_TYPE' => 'application/json',
+        ], '{"name": "Light", "description": "Light weight items"}');
+
+        $response = $this->client->getResponse();
+        $this->assertResponseCode($response, 204);
+
+        $this->client->request('GET', '/api/shipping-categories/'.$id, [], [], [
+            'HTTP_Authorization' => 'Bearer access_token',
+            'ACCEPT' => 'application/json',
+        ]);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'shipping_category/updated_shipping_category', 200);
+    }
+
+    public function testPartialUpdateShippingCategoryWhichDoesNotExistResponse()
+    {
+        $this->client->request('PATCH', '/api/shipping-categories/-1', [], [], [
+            'HTTP_Authorization' => 'Bearer access_token',
+            'ACCEPT' => 'application/json',
+        ]);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'error/not_found', 404);
+    }
+
+    public function testPartialUpdateShippingCategoryResponse()
+    {
+        $this->resetDataBase();
+        $id = $this->createShippingCategory();
+
+        $this->client->request('PATCH', '/api/shipping-categories/'.$id, [], [], [
+            'HTTP_Authorization' => 'Bearer access_token',
+            'CONTENT_TYPE' => 'application/json',
+        ], '{"name": "Light", "description": "Light weight items"}');
+
+        $response = $this->client->getResponse();
+        $this->assertResponseCode($response, 204);
+
+        $this->client->request('GET', '/api/shipping-categories/'.$id, [], [], [
+            'HTTP_Authorization' => 'Bearer access_token',
+            'ACCEPT' => 'application/json',
+        ]);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'shipping_category/updated_shipping_category', 200);
+    }
+
+    private function resetDataBase()
+    {
+        $this->purgeDataBase();
+        $this->loadPlatformData();
```

Comment:
``loadPlatformData`` is a weird name. Why not ``purgeDatabase`` and ``loadFixtures``?